src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -218,14 +220,69 @@ public IPAddressInformation IP
             }
         }
 
-        private void GetTxtRecord(out ushort length, out byte[] value)
+        /// <summary>
+        /// Returns TXT records in the form of key and value.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// </remarks>
+        /// <returns>returns empty dictionary in case of empty txt record else returns the key and value dictionary</returns>
+        /// <since_tizen> 9 </since_tizen>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public IDictionary<string, string> GetTxtRecord()
         {
-            int ret = Interop.Nsd.Dnssd.GetAllTxtRecord(_serviceHandle, out length, out value);
+            var map = new Dictionary<string, string>();
+
+            byte[] data = GetRawTxtRecord();
+            if (data.Length > 0)
+            {
+                var index = 0;
+                while (index < data.Length)
+                {
+                    var txtLen = data[index++];
+                    if (txtLen + index > data.Length)
+                    {
+                        Log.Error(Globals.LogTag, "Invalid data");
+                        NsdErrorFactory.ThrowDnssdException((int)DnssdError.InvalidOperation);
+                    }
+
+                    string record = Encoding.UTF8.GetString(data, index, txtLen);
+                    string[] list = record.Split('=');
+                    map.Add(list[0], list[1]);
+                    index += txtLen;
+                }
+            }
+            return map;
+        }
+
+        /// <summary>
+        /// Returns all raw TXT records.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// </remarks>
+        /// <returns>Returns empty bytes in case of no records available else returns raw text tecords.</returns>
+        /// <since_tizen> 9 </since_tizen>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public byte[] GetRawTxtRecord()
+        {
+            int ret = Interop.Nsd.Dnssd.GetAllTxtRecord(_serviceHandle, out ushort length, out IntPtr data);
```

Comment:
Who is release data?

>     You must release value using free(). 
https://docs.tizen.org/application/native/api/mobile/5.5/group__CAPI__NETWORK__DNSSD__MODULE.html#ga0cd3205024d1312e66bbaecd37eea03a